### PR TITLE
Populate database based on code

### PR DESCRIPTION
**Pull request by svenvanderburg**
Fixes https://github.com/PerfectFit-project/virtual-coach-issues/issues/158

## Changes in this PR:
* Remove `test.dump`, we will not make use of dumps anymore.
* Add code for populating the database with test data `helper/populate_db.py`
* Add unittests that test the code that adds test data. This tests whether the test data matches the sqlalchemy models,  thereby also covering the code in models.py. Also we check if the test data fills all tables with at least one row.
* Add github action for running unit tests
* Remove util scripts for making and restoring dumps

### CODE REVIEW

**Comment by dsmits** [comment]
I don't think you should add a `__init__.py` to the `tests` dir.
`tests` is not a package, it is just a regular dir that contains test scripts.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        which python3
+        python3 --version
+    - name: Upgrade pip and install dependencies
+      run: |
+        python3 -m pip install --upgrade pip setuptools
+        python3 -m pip install -r requirements.txt
+        python3 -m pip install -r requirements-dev.txt
+    - name: Run unit tests
+      run: |
+        export PYTHONPATH=$PYTHONPATH:$PWD
```
Comment (CunliangGeng) [inline .github/workflows/test.yml:34]: is this `export` necessary? my local test does not need it

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+from sqlalchemy import create_engine
```
Comment (CunliangGeng) [inline tests/conftest.py:1]: it'd be nice to add a comment to mention this file is a pytest configure file :-)

**Review by CunliangGeng — APPROVED** [review]
Looks good to me 👍

File: tests/conftest.py (hunk 2 above)
Comment (svenvanderburg) [reply tests/conftest.py:4]: Yes I agree, if you are not familiar with pytest this is confusing. Added a doctsring on top of the file.

File: .github/workflows/test.yml (hunk 1 above)
Comment (svenvanderburg) [reply .github/workflows/test.yml:34]: I tried, and it is indeed necessary :) See https://github.com/PerfectFit-project/virtual-coach-db/runs/4702749002?check_suite_focus=true

**Comment by svenvanderburg** [comment]
You are right @dsmits  I removed `__init__.py` from `tests` :)

File: .github/workflows/test.yml (hunk 1 above)
Comment (CunliangGeng) [reply .github/workflows/test.yml:34]: then it's necessary :-)

**Pull request merged**